### PR TITLE
Add `information_schema.columns.extra` and `information_schema.tables.table_rows`

### LINKED ISSUE #55182: `information_schema.tables` lacks field `TABLE_ROWS` (MySQL compatibility)
Required for Tableau Online.

Missing columns: 'TABLE_ROWS' while processing query:
```sql
SELECT TABLE_ROWS
FROM (SELECT database                                                 AS table_catalog,
             database                                                 AS table_schema,
             name                                                     AS table_name,
             multiIf(is_temporary, 'LOCAL TEMPORARY', engine LIKE '%View', 'VIEW', engine LIKE 'System%', 'SYSTEM VIEW',
                     has_own_data = 0, 'FOREIGN TABLE', 'BASE TABLE') AS table_type,
             total_bytes                                              AS data_length,
             'utf8mb4_0900_ai_ci'                                     AS table_collation,
             comment                                                  AS table_comment,
             table_catalog                                            AS TABLE_CATALOG,
             table_schema                                             AS TABLE_SCHEMA,
             table_name                                               AS TABLE_NAME,
             table_type                                               AS TABLE_TYPE,
             data_length                                              AS DATA_LENGTH,
             table_collation                                          AS TABLE_COLLATION,
             table_comment                                            AS TABLE_COMMENT
      FROM system.tables
      HAVING (table_name = 'commits')
         AND (table_schema = 'default')) AS TABLES
WHERE (table_schema = 'default')
  AND (table_name = 'commits')
```

**Describe the unexpected behaviour**
`TABLE_ROWS` column is available in both lowercase and uppercase variants. Probably it will be also good to implement the rest of the default MySQL columns as well.

**How to reproduce**
* Which ClickHouse server version to use: master
* Which interface to use, if matters: MySQL

CC @rschu1ze

### LINKED ISSUE #55183: `information_schema.columns` lacks field `EXTRA` (MySQL compatibility)
It is specifically required for Tableau _Desktop_ via MySQL, as one of the popular use cases is seamless data import/export between Tableau Desktop and Tableau Online using the same connector (MySQL in this case).

**Describe the unexpected behaviour**

This generated query:

```sql
SELECT TABLE_SCHEMA,
       NULL,
       TABLE_NAME,
       COLUMN_NAME,
       multiIf(upper(DATA_TYPE) = 'DECIMAL', 3, upper(DATA_TYPE) = 'DECIMAL UNSIGNED', 3, upper(DATA_TYPE) = 'TINYINT',
               if((position('ZEROFILL', upper(COLUMN_TYPE)) = 0) AND (position('UNSIGNED', upper(COLUMN_TYPE)) = 0) AND
                  (position('(1)', COLUMN_TYPE) != 0), -7, -6), upper(DATA_TYPE) = 'TINYINT UNSIGNED', if(
                           (position('ZEROFILL', upper(COLUMN_TYPE)) = 0) AND
                           (position('UNSIGNED', upper(COLUMN_TYPE)) = 0) AND (position('(1)', COLUMN_TYPE) != 0), -7,
                           -6), upper(DATA_TYPE) = 'BOOLEAN', 16, upper(DATA_TYPE) = 'SMALLINT', 5,
               upper(DATA_TYPE) = 'SMALLINT UNSIGNED', 5, upper(DATA_TYPE) = 'INT', 4,
               upper(DATA_TYPE) = 'INT UNSIGNED', 4, upper(DATA_TYPE) = 'FLOAT', 7, upper(DATA_TYPE) = 'FLOAT UNSIGNED',
               7, upper(DATA_TYPE) = 'DOUBLE', 8, upper(DATA_TYPE) = 'DOUBLE UNSIGNED', 8, upper(DATA_TYPE) = 'NULL', 0,
               upper(DATA_TYPE) = 'TIMESTAMP', 93, upper(DATA_TYPE) = 'BIGINT', -5,
               upper(DATA_TYPE) = 'BIGINT UNSIGNED', -5, upper(DATA_TYPE) = 'MEDIUMINT', 4,
               upper(DATA_TYPE) = 'MEDIUMINT UNSIGNED', 4, upper(DATA_TYPE) = 'DATE', 91, upper(DATA_TYPE) = 'TIME', 92,
               upper(DATA_TYPE) = 'DATETIME', 93, upper(DATA_TYPE) = 'YEAR', 91, upper(DATA_TYPE) = 'VARCHAR', 12,
               upper(DATA_TYPE) = 'VARBINARY', -3, upper(DATA_TYPE) = 'BIT', -7, upper(DATA_TYPE) = 'JSON', -1,
               upper(DATA_TYPE) = 'ENUM', 1, upper(DATA_TYPE) = 'SET', 1, upper(DATA_TYPE) = 'TINYBLOB', -3,
               upper(DATA_TYPE) = 'TINYTEXT', 12, upper(DATA_TYPE) = 'MEDIUMBLOB', -4, upper(DATA_TYPE) = 'MEDIUMTEXT',
               -1, upper(DATA_TYPE) = 'LONGBLOB', -4, upper(DATA_TYPE) = 'LONGTEXT', -1, upper(DATA_TYPE) = 'BLOB', -4,
               upper(DATA_TYPE) = 'TEXT', -1, upper(DATA_TYPE) = 'CHAR', 1, upper(DATA_TYPE) = 'BINARY', -2,
               upper(DATA_TYPE) = 'GEOMETRY', -2, upper(DATA_TYPE) = 'UNKNOWN', 1111, upper(DATA_TYPE) = 'POINT', -2,
               upper(DATA_TYPE) = 'LINESTRING', -2, upper(DATA_TYPE) = 'POLYGON', -2, upper(DATA_TYPE) = 'MULTIPOINT',
               -2, upper(DATA_TYPE) = 'MULTILINESTRING', -2, upper(DATA_TYPE) = 'MULTIPOLYGON', -2,
               upper(DATA_TYPE) = 'GEOMETRYCOLLECTION', -2, upper(DATA_TYPE) = 'GEOMCOLLECTION', -2, 1111) AS DATA_TYPE,
       upper(multiIf(upper(DATA_TYPE) = 'TINYINT', multiIf(
                   (position('ZEROFILL', upper(COLUMN_TYPE)) = 0) AND (position('UNSIGNED', upper(COLUMN_TYPE)) = 0) AND
                   (position('(1)', COLUMN_TYPE) != 0), 'BIT',
                   (position('UNSIGNED', upper(COLUMN_TYPE)) != 0) AND (position('UNSIGNED', upper(DATA_TYPE)) = 0),
                   'TINYINT UNSIGNED', DATA_TYPE), (position('UNSIGNED', upper(COLUMN_TYPE)) != 0) AND
                                                   (position('UNSIGNED', upper(DATA_TYPE)) = 0) AND
                                                   (position('SET', upper(DATA_TYPE)) != 1) AND
                                                   (position('ENUM', upper(DATA_TYPE)) != 1),
                     concat(DATA_TYPE, ' UNSIGNED'), upper(DATA_TYPE) = 'POINT', 'GEOMETRY',
                     upper(DATA_TYPE) = 'LINESTRING', 'GEOMETRY', upper(DATA_TYPE) = 'POLYGON', 'GEOMETRY',
                     upper(DATA_TYPE) = 'MULTIPOINT', 'GEOMETRY', upper(DATA_TYPE) = 'MULTILINESTRING', 'GEOMETRY',
                     upper(DATA_TYPE) = 'MULTIPOLYGON', 'GEOMETRY', upper(DATA_TYPE) = 'GEOMETRYCOLLECTION', 'GEOMETRY',
                     upper(DATA_TYPE) = 'GEOMCOLLECTION', 'GEOMETRY', upper(DATA_TYPE)))                   AS TYPE_NAME,
       upper(multiIf(upper(DATA_TYPE) = 'DATE', 10, upper(DATA_TYPE) = 'TIME',
                     8 + if(DATETIME_PRECISION > 0, DATETIME_PRECISION + 1, DATETIME_PRECISION),
                     (upper(DATA_TYPE) = 'DATETIME') OR (upper(DATA_TYPE) = 'TIMESTAMP'),
                     19 + if(DATETIME_PRECISION > 0, DATETIME_PRECISION + 1, DATETIME_PRECISION),
                     upper(DATA_TYPE) = 'YEAR', 4,
                     (upper(DATA_TYPE) = 'TINYINT') AND (position('ZEROFILL', upper(COLUMN_TYPE)) = 0) AND
                     (position('UNSIGNED', upper(COLUMN_TYPE)) = 0) AND (position('(1)', COLUMN_TYPE) != 0), 1,
                     (upper(DATA_TYPE) = 'MEDIUMINT') AND (position('UNSIGNED', upper(COLUMN_TYPE)) != 0), 8,
                     upper(DATA_TYPE) = 'JSON', 1073741824, upper(DATA_TYPE) = 'GEOMETRY', 65535,
                     upper(DATA_TYPE) = 'POINT', 65535, upper(DATA_TYPE) = 'LINESTRING', 65535,
                     upper(DATA_TYPE) = 'POLYGON', 65535, upper(DATA_TYPE) = 'MULTIPOINT', 65535,
                     upper(DATA_TYPE) = 'MULTILINESTRING', 65535, upper(DATA_TYPE) = 'MULTIPOLYGON', 65535,
                     upper(DATA_TYPE) = 'GEOMETRYCOLLECTION', 65535, upper(DATA_TYPE) = 'GEOMCOLLECTION', 65535,
                     CHARACTER_MAXIMUM_LENGTH IS NULL, NUMERIC_PRECISION, CHARACTER_MAXIMUM_LENGTH > 2147483647,
                     2147483647,
                     CHARACTER_MAXIMUM_LENGTH))                                                            AS COLUMN_SIZE,
       65535                                                                                               AS BUFFER_LENGTH,
       upper(multiIf(upper(DATA_TYPE) = 'DECIMAL', NUMERIC_SCALE,
                     (upper(DATA_TYPE) = 'FLOAT') OR (upper(DATA_TYPE) = 'DOUBLE'),
                     if(NUMERIC_SCALE IS NULL, 0, NUMERIC_SCALE),
                     NULL))                                                                                AS DECIMAL_DIGITS,
       10                                                                                                  AS NUM_PREC_RADIX,
       if(IS_NULLABLE = 'NO', 0, if(IS_NULLABLE = 'YES', 1, 2))                                            AS NULLABLE,
       COLUMN_COMMENT                                                                                      AS REMARKS,
       COLUMN_DEFAULT                                                                                      AS COLUMN_DEF,
       0                                                                                                   AS SQL_DATA_TYPE,
       0                                                                                                   AS SQL_DATETIME_SUB,
       if(CHARACTER_OCTET_LENGTH > 2147483647, 2147483647,
          CHARACTER_OCTET_LENGTH)                                                                          AS CHAR_OCTET_LENGTH,
       ORDINAL_POSITION,
       IS_NULLABLE,
       NULL                                                                                                AS SCOPE_CATALOG,
       NULL                                                                                                AS SCOPE_SCHEMA,
       NULL                                                                                                AS SCOPE_TABLE,
       NULL                                                                                                AS SOURCE_DATA_TYPE,
       if(EXTRA LIKE '%auto_increment%', 'YES', 'NO')                                                      AS IS_AUTOINCREMENT,
       if(EXTRA LIKE '%GENERATED%', 'YES', 'NO')                                                           AS IS_GENERATEDCOLUMN
FROM (SELECT database                 AS table_catalog,
             database                 AS table_schema,
             table                    AS table_name,
             name                     AS column_name,
             position                 AS ordinal_position,
             default_expression       AS column_default,
             type LIKE 'Nullable(%)'  AS is_nullable,
             type                     AS data_type,
             character_octet_length   AS character_maximum_length,
             character_octet_length,
             numeric_precision,
             numeric_precision_radix,
             numeric_scale,
             datetime_precision,
             NULL                     AS character_set_catalog,
             NULL                     AS character_set_schema,
             NULL                     AS character_set_name,
             NULL                     AS collation_catalog,
             NULL                     AS collation_schema,
             NULL                     AS collation_name,
             NULL                     AS domain_catalog,
             NULL                     AS domain_schema,
             NULL                     AS domain_name,
             comment                  AS column_comment,
             type                     AS column_type,
             table_catalog            AS TABLE_CATALOG,
             table_schema             AS TABLE_SCHEMA,
             table_name               AS TABLE_NAME,
             column_name              AS COLUMN_NAME,
             ordinal_position         AS ORDINAL_POSITION,
             column_default           AS COLUMN_DEFAULT,
             is_nullable              AS IS_NULLABLE,
             data_type                AS DATA_TYPE,
             character_maximum_length AS CHARACTER_MAXIMUM_LENGTH,
             character_octet_length   AS CHARACTER_OCTET_LENGTH,
             numeric_precision        AS NUMERIC_PRECISION,
             numeric_precision_radix  AS NUMERIC_PRECISION_RADIX,
             numeric_scale            AS NUMERIC_SCALE,
             datetime_precision       AS DATETIME_PRECISION,
             character_set_catalog    AS CHARACTER_SET_CATALOG,
             character_set_schema     AS CHARACTER_SET_SCHEMA,
             character_set_name       AS CHARACTER_SET_NAME,
             collation_catalog        AS COLLATION_CATALOG,
             collation_schema         AS COLLATION_SCHEMA,
             collation_name           AS COLLATION_NAME,
             domain_catalog           AS DOMAIN_CATALOG,
             domain_schema            AS DOMAIN_SCHEMA,
             domain_name              AS DOMAIN_NAME,
             column_comment           AS COLUMN_COMMENT,
             column_type              AS COLUMN_TYPE
      FROM system.columns
      HAVING (COLUMN_NAME LIKE '%')
         AND ((TABLE_NAME = 'commits') AND (TABLE_SCHEMA = 'default'))) AS COLUMNS
WHERE (TABLE_SCHEMA = 'default')
  AND (TABLE_NAME = 'commits')
  AND (COLUMN_NAME LIKE '%')
ORDER BY TABLE_SCHEMA ASC, TABLE_NAME ASC, ORDINAL_POSITION ASC
```

fails with 

```
Code: 47. DB::Exception: Missing columns: 'EXTRA' while processing query
```

**How to reproduce**
* Which ClickHouse server version to use: master
* Which interface to use, if matters: MySQL

**Expected behavior**
`information_schema.columns` has the `EXTRA` column in lowercase and uppercase. It will probably also be good to implement the rest of the default MySQL columns as well. 

CC @rschu1ze

### PR DESCRIPTION
**Pull request by rschu1ze**
Fixes: #55183
Fixes: #55182

### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
To improve compatibility with MySQL, 1. "information_schema.tables" now includes the new field "table_rows", and 2. "information_schema.columns" now includes the new field "extra".

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 907e792a932c30224cf858e9718924e206f7aa93 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55215/907e792a932c30224cf858e9718924e206f7aa93/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6393504526">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55215/907e792a932c30224cf858e9718924e206f7aa93/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by rschu1ze** [comment]
@slvrtrn Like to approve this one?

File: docs/en/operations/system-tables/information_schema.md [hunk 1]
```diff
@@ -69,6 +69,7 @@ Columns:
 - `domain_catalog` ([Nullable](../../sql-reference/data-types/nullable.md)([String](../../sql-reference/data-types/string.md))) — `NULL`, not supported.
 - `domain_schema` ([Nullable](../../sql-reference/data-types/nullable.md)([String](../../sql-reference/data-types/string.md))) — `NULL`, not supported.
 - `domain_name` ([Nullable](../../sql-reference/data-types/nullable.md)([String](../../sql-reference/data-types/string.md))) — `NULL`, not supported.
+- `extras` ([Nullable](../../sql-reference/data-types/nullable.md)([String](../../sql-reference/data-types/string.md))) — `STORED GENERATED` for `MATERIALIZED`-type columns, `VIRTUAL GENERATED` for `ALIAS`-type columns, `DEFAULT_GENERATED` for `DEFAULT`-type columns, or `NULL`.
```
Comment (slvrtrn) [inline docs/en/operations/system-tables/information_schema.md:72]: Is it `extras`, though? I see it is named `extra` in the [docs](https://dev.mysql.com/doc/refman/8.0/en/information-schema-columns-table.html)

File: docs/en/operations/system-tables/information_schema.md [hunk 2]
```diff
@@ -198,6 +199,8 @@ Columns:
     - `FOREIGN TABLE`
     - `LOCAL TEMPORARY`
     - `SYSTEM VIEW`
+- `data_rows` ([Nullable](../../sql-reference/data-types/nullable.md)([UInt64](../../sql-reference/data-types/int-uint.md))) — The total
```
Comment (slvrtrn) [inline docs/en/operations/system-tables/information_schema.md:202]: `table_rows`? See the [docs](https://dev.mysql.com/doc/mysql-infoschema-excerpt/8.0/en/information-schema-tables-table.html)

File: docs/en/operations/system-tables/information_schema.md (hunk 1 above)
Comment (rschu1ze) [reply docs/en/operations/system-tables/information_schema.md:72]: Good find, fixed.

File: docs/en/operations/system-tables/information_schema.md (hunk 2 above)
Comment (rschu1ze) [reply docs/en/operations/system-tables/information_schema.md:202]: Also fixed, tnx.

**Review by slvrtrn — APPROVED** [review]

**Pull request merged**